src/App.tsx

```diff
@@ -1,8 +1,96 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './App.css';
+import { getGoods } from './Api';
+import { GoodsList } from './components/GoodsList';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface AppState {
+  goodsList: Goods;
+  isLoaded: boolean;
+}
+
+class App extends Component<{}, AppState> {
+  state = {
+    goodsList: [],
+    isLoaded: false,
+  };
+
+  loadGoods = (event: React.MouseEvent<HTMLButtonElement>) => {
+    const { name } = event.currentTarget.dataset;
```

Comment:
Do not use data-attribi=utes. You do not need it, pass values as arguments to the function